plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/PlayAssetDeliveryTestHelper.java

```diff
@@ -46,4 +53,45 @@ public static Dictionary createAssetPackLocationsDictionary() {
     testDict.put("location2", innerDict2);
     return testDict;
   }
+
+  /**
+   * Mock object factory that returns a mock Task<T> object. Will invoke onSuccessListener with
+   * result if addOnSuccessListener() is called.
+   *
+   * @param result object to be passed to the onSuccessListener
+   * @param <T> parameterized type of the Task object
+   * @return instantiated mock Task
+   */
+  public static <T> Task<T> createMockOnSuccessTask(T result) {
+    Task<T> returnTaskMock = mock(Task.class);
+    doAnswer(
+            invocation -> {
+              OnSuccessListener<T> listener = (OnSuccessListener<T>) invocation.getArguments()[0];
+              listener.onSuccess(result);
+              return null;
+            })
+        .when(returnTaskMock)
+        .addOnSuccessListener(any(OnSuccessListener.class));
+    return returnTaskMock;
+  }
+
+  /**
+   * Mock object factory that returns a mock Task<T> object. Will invoke onFailureListener with
+   * Exception if addOnFailureListener() is called.
+   *
+   * @param <T> parameterized type of the Task object
+   * @return instantiated mock Task
+   */
+  public static <T> Task<T> createMockOnFailureTask() {
+    Task<T> returnTaskMock = mock(Task.class);
+    doAnswer(
+            invocation -> {
+              OnFailureListener listener = (OnFailureListener) invocation.getArguments()[0];
+              listener.onFailure(new Exception("Test Exception!"));
```

Comment:
Rather than hard coding the exception, I would provide the Exception object as an argument, like we do with the `createMockOnSuccessTask` result. This way tests don't have to rely on an arbitrary implementation detail (i.e. `"Test Exception!"`) to verify that the correct exception was thrown.